cldfbench_doreco.py

```diff
@@ -15,21 +20,46 @@ def cldf_specs(self):  # A dataset must declare all CLDF sets it creates.
                 module='StructureDataset'
                 )
 
+    def cmd_download(self, args):
+        self.raw_dir.download(
+            'https://sharedocs.huma-num.fr/wl/?id=s947TcfRfDZR643QURdyncdku4EmeKyb&fmode=download',
+            'languages.csv')
+        self.raw_dir.download(
+            'https://sharedocs.huma-num.fr/wl/?id=xqkUR3WoFOKB8cRb0MdHYJxIDEXEXlVG&fmode=download',
+            'sources.bib')
+
+        for row in self.raw_dir.read_csv('languages.csv', dicts=True):
+            if 'ND' not in row['Annotation license']:
+                print(row['Glottocode'], row['DOI'])
+                cid = urllib.parse.quote(row['DOI'], safe='')
+                for f in requests.get("{}datas/{}?metadata-format=dc".format(NAKALA_API, cid)).json()['files']:
+                    for s in ['_wd.csv', '_ph.csv', '_metadata.csv', '_gloss-abbreviations.csv']:
+                        if f['name'].endswith(s):
+                            print(f['name'], f['sha1'])
+                            self.raw_dir.download('{}data/{}/{}'.format(NAKALA_API, cid, f['sha1']), '{}{}'.format(row['Glottocode'], s))
+                            break
+
+    def iter_rows(self, pattern):
```

Comment:
Here, we "merge" the files of a certain type across corpora.

Reply:
Yup, that's way more efficient than what I had done.